src/state-schemas/state-schema.js

```diff
@@ -229,11 +229,17 @@ export const stateProperties = {
       }
     }
   },
+  checkingRevisionIsLatest: {
```

Comment:
via ctrl-F, I'm spotting where this piece of state is set, but not where it is actually used. Maybe this is left over from an earlier idea?

Reply:
good catch, I think there's actually some bugs here (or at least poorly specified behaviour). looking into it...

Reply:
ok np. i'll hold off on more review until you give me the go ahead. thanks @wlach